Inferences/Induction.cpp

```diff
@@ -63,11 +63,46 @@ TermList TermReplacement::transformSubterm(TermList trm)
   CALL("TermReplacement::transformSubterm");
 
   if(trm.isTerm() && trm.term()==_o){
-   return _r;
+    return _r;
   }
   return trm;
 }
 
+TermList LiteralSubsetReplacement::transformSubterm(TermList trm)
+{
+  CALL("LiteralSubsetReplacement::transformSubterm");
+
+  if(trm.isTerm() && trm.term() == _o){
+    // Replace either if there are too many occurrences to try all possibilities,
+    // or if the bit in _iteration corresponding to this match is set to 1.
+    if ((_occurrences > _maxOccurrences) || (1 & (_iteration >> _matchCount++))) {
+      return _r;
+    }
+  }
+  return trm;
+}
+
+Literal* LiteralSubsetReplacement::transformSubset() {
+  CALL("LiteralSubsetReplacement::transformSubset");
+  // Increment _iteration, since it either is 0, or was already used.
+  _iteration++;
+  static unsigned maxSubsetSize = env.options->maxInductionTermSubsetSize();
+  int setBits = __builtin_popcount(_iteration);
```

Comment:
Add a comment about compiler support - I don't think it's currently an issue but will help somebody debug in the future if it becomes an issue.

Reply:
Good point, I added a comment.